guppylang/tracing/function.py

```diff
@@ -152,7 +152,10 @@ def trace_call(func: CompiledCallableDef, *args: Any) -> Any:
     ]
 
     # Create dummy variables and bind the objects to them
-    arg_vars = [Variable(next(tmp_vars), obj._ty, None) for obj in args_objs]
+    arg_vars: list[Variable] = [
+        ComptimeVariable(next(tmp_vars), obj._ty, None, static_value=arg)
+        for (obj, arg) in zip(args_objs, args, strict=False)
```

Comment:
Why `strict=False`?

Reply:
I think I didn't add it at all and the check automatically added, changed it to True